scispacy/data_util.py

```diff
@@ -61,18 +61,69 @@ def med_mentions_example_iterator(filename: str) -> Iterator[MedMentionExample]:
         if lines:
             yield process_example(lines)
 
-def read_med_mentions(filename: str):
+def remove_overlapping_entities(sorted_spacy_format_entities: List[Tuple[int, int, str]]
+                                ) -> List[Tuple[int, int, str]]:
     """
-    Reads in the MedMentions dataset into Spacy's
-    NER format.
+    Removes overlapping entities from the entity set, by greedilytaking the longest
+    entity from each overlapping chain. The input list of entities should be sorted
+    and follow the spacy format.
     """
-    examples = []
-    for example in med_mentions_example_iterator(filename):
-        spacy_format_entities = [(x.start, x.end, x.mention_type) for x in example.entities]
-        examples.append((example.text, {"entities": spacy_format_entities}))
-
-    return examples
+    entity_index = 0
+    spacy_format_entities_without_overlap = []
+    stack = []
+    stack_start = 0
+    stack_end = 0
+    while True:
```

Comment:
This while loop should be a for loop, because you don't modify the `entity_index` variable apart from at the end of the loop. You can then simplify the logic in the loop by handling the `current_entity=None` case outside the body of the loop.